docker/constants.py

```diff
@@ -0,0 +1,32 @@
+from enum import Enum
+import os
+ACTION_INVOKED_GROUP = "actions.action.invoked"
+TIPS_GROUP = "ui.tips.tip.shown"
+FORGET_TIME_DAYS = 10
+
+
+class Method(Enum):
+    TOP = 0
+    PROB = 1
+    MATRIX = 2
+
+
+METHODS_CNT = 3
```

Comment:
Shouldn't we consider allocating more groups to avoid changing behavior after adding new models? In the current implementation, adding one more model shifts users groups. I'd suggest allocating 8 or 16 groups and then change behavior only for a few of them. After adding more methods we'll just start to predict for one to many untouched groups without affecting existing ones.

Reply:
Ok. And what should be returned for the groups without implementation?

Reply:
Randomly shuffled tips? I use `"random_shuffle"` string to specify the algorithm name if the server is inaccessible. You can use the same value, or `server_random_shuffle` to distinguish these two cases.

Reply:
Done.